### DISCUSSION TITLE
Set up a progressive number

### DISCUSSION THREAD

**Discussion by vlakius** [tortoise/tortoise-orm#1255] [Q&A] [answered]
Hello everyone, 
I am trying to create a column called "progressive" whose value should be auto generated.

The ideal format it should have would be "id/year".
Where id is the primary key of the row and year is the 'current year 
I tried overriding the save method as follows:

  class Test(Model):
      id = fields.IntField(pk=True)
      date = fields.DatetimeField()
      name = fields.CharField()
      progressive = fields.CharField()

      async def save(self, *args, **kwargs):
          self.progressive = f"{self.id}/{self.date.year}"
          return super().save(*args, **kwargs)

but unfortunately when I save the progressive field it comes up "None/2022". I guess this is because at the time of saving there is no way to know what the 'id' of the record will be. 

I don't have to use the record id as long as the field is unique and auto generated. Is there any workaround ?

**Comment by long2ice** [comment, marked as answer]
What about use `signals.post_save`?

**Reply by vlakius** [reply]
Thank you so much for the tip.
For those who had the same problem here's how I 'solved it
```
class Test(Model):
id = fields.IntField(pk=True)
date = fields.DatetimeField()
name = fields.CharField(max_length=40)
progressive = fields.CharField(max_length=40, null=True)   # <------- null = True is important 


@post_save(Test)
async def create_progress(sender, instance, created, using_db, update_fields, **kwargs):
    if created:
        instance.progressive = f"{instance.id}/{instance.date.year}"
        await instance.save()
```

If there is a more elegant solution let me know.
